packages/dataviews/src/components/dataform/stories/index.story.tsx

```diff
@@ -486,3 +490,276 @@ export const Visibility = {
+			type: 'text',
+		},
+		{
+			id: 'billingAddress',
+			label: 'Billing Address',
+			type: 'text',
+		},
+		{
+			id: 'displayPayments',
+			label: 'Display Payments?',
+			type: 'boolean',
+		},
+		{
+			id: 'payments',
+			label: 'Payments',
+			type: 'text',
+			readOnly: true, // Triggers using the render method instead of Edit.
+			isVisible: ( item ) => item.displayPayments,
+			render: ( { item } ) => {
+				return (
+					<p>
+						The customer has made a total of { item.totalOrders }{ ' ' }
+						orders, amounting to { item.totalRevenue } dollars. The
+						average order value is { item.averageOrderValue }{ ' ' }
+						dollars.
+					</p>
+				);
+			},
+		},
+		{
+			id: 'vat',
+			label: 'VAT',
+			type: 'integer',
+		},
+		{
+			id: 'commission',
+			label: 'Commission',
+			type: 'integer',
+		},
+	];
+
+	const [ customer, setCustomer ] = useState< Customer >( {
+		name: 'Danyka Romaguera',
+		email: 'aromaguera@example.org',
+		phone: '1-828-352-1250',
+		plan: 'Business',
+		shippingAddress: 'N/A',
+		billingAddress: 'Danyka Romaguera, West Myrtiehaven, 80240-4282, BI',
+		displayPayments: true,
+		totalOrders: 2,
+		totalRevenue: 1430,
+		averageOrderValue: 715,
+		hasVat: true,
+		vat: 10,
+		commission: 5,
+	} );
+
+	const form = useMemo(
+		() =>
+			( {
+				layout: {
+					type: 'card',
+					// labelPosition: 'none',
+					// TODO: fix labelPosition none.
+					// When this is set, the second card doesn't get the header (even if it doesn't have labelPosition set),
+					// but the 1st and 3rd do. I'd expect none of them get the header.
+				},
+				fields: [
+					{
+						id: 'customerCard',
+						label: 'Customer',
+						children: [
+							{
+								id: 'customerContact',
+								label: 'Contact',
+								layout: { type: 'panel', labelPosition: 'top' },
+								children: [
+									{
+										id: 'name',
+										layout: {
+											type: 'regular',
+											labelPosition: 'top',
+										},
+									},
+									{
+										id: 'phone',
+										layout: {
+											type: 'regular',
+											labelPosition: 'top',
+										},
+									},
+									{
+										id: 'email',
+										layout: {
+											type: 'regular',
+											labelPosition: 'top',
+										},
+									},
+								],
+							},
+							{
+								id: 'plan',
+								layout: { type: 'panel', labelPosition: 'top' },
+							},
+							{
+								id: 'shippingAddress',
+								layout: { type: 'panel', labelPosition: 'top' },
+							},
+							{
+								id: 'billingAddress',
+								layout: { type: 'panel', labelPosition: 'top' },
+							},
+							'displayPayments',
+						],
+					},
+					{
+						id: 'payments',
+						layout: { type: 'card', labelPosition: 'none' },
+					},
+					{
+						id: 'taxConfiguration',
+						label: 'Taxes',
+						layout: {
+							type: 'card',
+							opened: false, // TODO: it does not work.
+						},
+						children: [
+							// TODO: default values.
```

Comment:
Related to the comment at https://github.com/WordPress/gutenberg/pull/71100/files#r2262380433